lib/edgehog_device/ota.c

```diff
@@ -474,6 +483,25 @@ static void ota_thread_entry_point(void *edgehog_device, void *ptr2, void *ptr3)
     edgehog_settings_save(OTA_KEY, OTA_STATE_KEY, &ota_state, sizeof(uint8_t));
 }
 
+static void wait_for_reboot(void)
+{
+#ifdef CONFIG_EDGEHOG_DEVICE_ZBUS_OTA_EVENT
+    k_timepoint_t wait_timepoint = sys_timepoint_calc(K_SECONDS(OTA_REBOOT_MAX_DELAY_S));
+    while (!K_TIMEOUT_EQ(sys_timepoint_timeout(wait_timepoint), K_NO_WAIT)) {
+        const struct zbus_channel *chan = NULL;
+        edgehog_ota_chan_event_t ota = { 0 };
+        if ((zbus_sub_wait(&edgehog_ota_internal_subscriber, &chan, K_SECONDS(1)) == 0)
+            && (&edgehog_ota_chan == chan) && (zbus_chan_read(chan, &ota, K_SECONDS(1)) == 0)
+            && (ota.event == EDGEHOG_OTA_CONFIRM_REBOOT_EVENT)) {
+            EDGEHOG_LOG_INF("Reboot confirmation received, skipping wait time.");
+            break;
+        }
+    }
+#else
+    k_sleep(K_SECONDS(OTA_REBOOT_MAX_DELAY_S));
```

Comment:
If the zbus event is disabled is there any point in waiting ?

Reply:
I think so, just to ensure all transmission/reception has been performed successfully.

Reply:
Mhh yes but I don't think it changes much since no event can be sent to the application it could have in flight messages even after a delay. I mean I don't think this will preserve messages better.

Reply:
Since we don't have a specific time to wait we add an acceptable time for a standard network connection to ensure all `OTA` transmission/reception has been performed successfully.

Reply:
As an alternative we could add a Kconfig option enabled only when the Zbus integration is disabled, this way the user can read that the option sets a "Default wait time to ensure that all OTA messages are correctly transmitted/received". And of course `OTA_REBOOT_MAX_DELAY_S` should be disabled when `CONFIG_EDGEHOG_DEVICE_ZBUS_OTA_EVENT` is disabled. The options could have different help messages and be easier to understand since they are different.

Reply:
If you want an alternative to `OTA_REBOOT_MAX_DELAY_S` you can already use Zbus to notify at your preferred time. Otherwise  Edgehog can wait  for `OTA_REBOOT_MAX_DELAY_S`  before reboot without adding another layer of configuration in Kconfig.